common/src/main/java/com/microsoft/identity/common/internal/ui/webview/AzureActiveDirectoryWebViewClient.java

```diff
@@ -77,13 +79,22 @@ public class AzureActiveDirectoryWebViewClient extends OAuth2WebViewClient {
     public static final String ERROR_SUBCODE = "error_subcode";
     public static final String ERROR_DESCRIPTION = "error_description";
     private final String mRedirectUrl;
+    private final IChallengeHandler<ClientCertRequest, Void> mClientCertAuthChallengeHandler;
 
     public AzureActiveDirectoryWebViewClient(@NonNull final Activity activity,
                                              @NonNull final IAuthorizationCompletionCallback completionCallback,
                                              @NonNull final OnPageLoadedCallback pageLoadedCallback,
                                              @NonNull final String redirectUrl) {
         super(activity, completionCallback, pageLoadedCallback);
         mRedirectUrl = redirectUrl;
+        //If api level is 19 or above, use EoClientCertAuthChallengeHander
+        //Otherwise, use regular ClientCertAuthChallengeHandler
+        if (android.os.Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
```

Comment:
ClientCertRequest object seems to be requiring API level 21... onReceivedClientCertRequest as well...

So.. I'm thinking that we might not need this check.

Can you try building MSAL test app with targetApi < 21 and observe the behavior?

Reply:
Tested with a device Api level 19 (KitKat).
Couldn't build msalTestApp: `Invoke-customs are only supported starting with Android O (--min-api 26)`
So I tried with AzureSample and the regular ClientCerthAuthChallengeHandler.
Sometimes the WebView crashes right away, but after the third try, I was able to try to sign in. When I try to login with a cert, the picker doesn't show up and it takes me directly to the error page. 
The logcat shows many DalvikVm errors, but the one pertaining to ClientCertAuthRequest is: `Could not find class 'android.webkit.ClientCertRequest', referenced from method com.microsoft.identity.common.internal.ui.webview.challengehandlers.ClientCertAuthChallengeHandler.processChallenge', referenced from method com.microsoft.identity.common.internal.ui.webview.challengehandlers.EoClientCertAuthChallengeHandler.processChallenge` 

This check is definitely not needed anymore, since it seems like any device <21 can't do regular CBA anyway.

Reply:
@melissaahn we should probably let vimala know (if request comes from android and x-client-OS < 21, do not trigger CBA)

cc @shoatman